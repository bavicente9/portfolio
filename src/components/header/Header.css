.header {
    position: sticky;
    top: 0;
    height: 50px;
}



.header__container {
    background: var(--GeneralBackGround);
    height: 50px;
    width: 100%;
    display: flex;
    justify-content: right;
    align-items: center;
    position: absolute;
    z-index: 500;

}




.button--navbar {
    height: 35px;
    margin-right: 25px;
    text-decoration: none;
    display: grid;
    place-items: center;
}

.header__navbar {
    padding-right: var(--GeneralMarginWidth);
    display: block;
}

.header__navbar ul {
    list-style: none;
    padding: 0;
    margin: 0;
    display: flex;
}

.header__navbar li a {
    width: 125px;
    height: 50px;
    display: grid;
    place-content: center;
    font-size: 1rem;
    cursor: pointer;

    text-decoration: none;
    font-family: 'Roboto', sans-serif;
    font-style: italic;
    color: rgb(255, 255, 255);
}

.header__navbar li:hover {
    color: #fff;
    background: linear-gradient(180deg, var(--DarkOrange) 0%, var(--VeryDarkOrange) 100%);
}



.header__buttonMenu {
    display: none;
}

.header__rMenu {
    display: none;
}


@media (max-width:775px) {
    .header__container {
        display: grid;
        grid-template-areas:
            'buttonContact buttonMenu'
        ;
        grid-template-columns: 1fr 35px;
        grid-template-rows: 50px;
        place-items: center end;
    }

    .header__navbar {
        display: none;
    }

    .button--navbar {
        grid-area: buttonContact;
        margin-right: calc(25px + var(--GeneralMarginWidth));
   
    }

    .header__buttonMenu {
        display: block;
        margin-right: var(--GeneralMarginWidth);
        background: url(./icon-hamburger.svg);
        background-size: 30px 30px;
        border: none;
        padding: 0;
        width: 30px;
        height: 30px;
        cursor: pointer;
        grid-area: buttonMenu;


    }

    .header__buttonMenu img {
        width: 100%;
        height: 100%;
    }

  
    /*Responsive menu with animation*/

    .header__rMenu {
        display: block;
        visibility: hidden;
        width: 100%;
        text-align: center;
        background: var(--GeneralBackGround);
        position: absolute;
        z-index: -1;
        top: -500%;
        color: #fff;


    }

    .header__rMenu--open {
        animation: rMenuAnimation--open 0.4s cubic-bezier(0.23, 1, 0.320, 1) 1 forwards;
    }

    .header__rMenu--close {
        animation: rMenuAnimation--close 0.4s cubic-bezier(0.23, 1, 0.320, 1) 1 forwards;
    }

    .header__rMenu ul {
        list-style: none;
        padding: 0;
        margin: 10px 0 0;
    }

    .header__rMenu li {
        height: min-content;
    }

    .header__rMenu li a {
        display: block;
        padding: 25px 0;
        border-top: 1px solid var(--Orange);
        font-size: 2rem;
        cursor: pointer;
        text-decoration: none;
        color: #fff;
        width: 100%;
    }
    
    .header__rMenu li:last-child{
        border-bottom: 1px solid var(--Orange);

    }

    .header__rMenu ul li:hover {
        background-color: var(--VeryDarkOrange);
    }

    @keyframes rMenuAnimation--open {
        from {
            top: -500%;
        }
        
        to {
            visibility: visible;
            top: 100%;
        }
    }

    @keyframes rMenuAnimation--close {
        from {
            top: 100%;
            visibility: visible;
        }

        to {
            top: -500%;
            visibility: hidden;
        }
    }


}