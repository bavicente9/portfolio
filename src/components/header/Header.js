import './Header.css'

export const Header = () => {

    //toggle visibility of the resoponsive menu
    const handleToggleRMenu = () => {
        const rMenu = document.getElementById('header__rMenu')

        if (!rMenu.classList.contains('header__rMenu--open')) {

            rMenu.classList.remove('header__rMenu--close')
            rMenu.classList.add('header__rMenu--open')


        } else {
            rMenu.classList.remove('header__rMenu--open')
            rMenu.classList.add('header__rMenu--close')

        }



    }
    return (
        <header aria-label='header' className='header'>
            <div className='header__container'>
                <a rel='tag' aria-label='Contact me' className='button button--navbar' href='#contact'>Contact</a>

                <nav role='menubar' id='header__navbar' className='header__navbar'>
                    <ul>
                        <li className='navbar__item'><a rel='section' href='#presentation'>Home</a></li>
                        <li className='navbar__item'><a rel='section' href='#projects' >Projects</a></li>
                        <li className='navbar__item'><a rel='section' href='#skills'>Skills</a></li>
                    </ul>
                </nav>

                <button  aria-label='toggle menu options'
                    className='header__buttonMenu buttonMenu--hidden buttonMenu--visible-true buttonMenu--visible-false'
                    onClick={handleToggleRMenu}>
                </button>

            </div>
            <nav role='menu' className='header__rMenu ' id='header__rMenu' >
                <ul>
                    <li className='navbar__item'><a rel='section' onClick={handleToggleRMenu} href='#presentation'>Home</a></li>
                    <li className='navbar__item'><a rel='section' onClick={handleToggleRMenu} href='#projects' >Projects</a></li>
                    <li className='navbar__item'><a rel='section' onClick={handleToggleRMenu} href='#skills'>Skills</a></li>
                </ul>
            </nav>
        </header>
    )
}