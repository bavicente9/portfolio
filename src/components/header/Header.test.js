import { Header } from "./Header"
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

describe('Testing Header', () => {

    it('renders correctly', () => {
        const { asFragment } = render(<Header />)
        expect(asFragment()).toMatchSnapshot()
    })

    it('responsive menu: open and close', () => {

        render(<Header></Header>)

        const rMenuOptions = screen.getByRole('menu')
        const hamburgerMenu = screen.getByRole('button');

        //check if the  hamburger button menu works correctly      
        //show options
        userEvent.click(hamburgerMenu)
        expect(rMenuOptions.classList.contains('header__rMenu--open')).toBe(true)
        //hidde options
        userEvent.click(hamburgerMenu)
        expect(rMenuOptions.classList.contains('header__rMenu--open')).toBe(false)
        expect(rMenuOptions.classList.contains('header__rMenu--close')).toBe(true)
    })
})