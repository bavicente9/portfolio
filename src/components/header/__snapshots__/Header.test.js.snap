// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Header renders correctly 1`] = `
<DocumentFragment>
  <header
    aria-label="header"
    class="header"
  >
    <div
      class="header__container"
    >
      <a
        aria-label="Contact me"
        class="button button--navbar"
        href="#contact"
        rel="tag"
      >
        Contact
      </a>
      <nav
        class="header__navbar"
        id="header__navbar"
        role="menubar"
      >
        <ul>
          <li
            class="navbar__item"
          >
            <a
              href="#presentation"
              rel="section"
            >
              Home
            </a>
          </li>
          <li
            class="navbar__item"
          >
            <a
              href="#projects"
              rel="section"
            >
              Projects
            </a>
          </li>
          <li
            class="navbar__item"
          >
            <a
              href="#skills"
              rel="section"
            >
              Skills
            </a>
          </li>
        </ul>
      </nav>
      <button
        aria-label="toggle menu options"
        class="header__buttonMenu buttonMenu--hidden buttonMenu--visible-true buttonMenu--visible-false"
      />
    </div>
    <nav
      class="header__rMenu "
      id="header__rMenu"
      role="menu"
    >
      <ul>
        <li
          class="navbar__item"
        >
          <a
            href="#presentation"
            rel="section"
          >
            Home
          </a>
        </li>
        <li
          class="navbar__item"
        >
          <a
            href="#projects"
            rel="section"
          >
            Projects
          </a>
        </li>
        <li
          class="navbar__item"
        >
          <a
            href="#skills"
            rel="section"
          >
            Skills
          </a>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
